kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -140,8 +144,19 @@ public KeyValueIterator<Bytes, byte[]> range(
       final Bytes to,
       final long minValidTs
   ) {
-    // NOTE: Make sure to handle filtering of tombstones in the remote range scan.
-    throw new UnsupportedOperationException();
+    final FindIterable<KVDoc> result = docs.find(
```

Comment:
I suspect tests for this are in your FK join PR?

Reply:
I didn't see that we had kvtable unit tests. I'm adding some cases now.